src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -2875,13 +2881,9 @@ public void AssertNoAttachmentsForDocument(DocumentsOperationContext context, st
                     throw new InvalidOperationException($"Found attachment {attachment.Name} but it should be deleted.");
                 }
 
-                var tree = context.Transaction.InnerTransaction.CreateTree(AttachmentsSlice);
-                using (var it = tree.Iterate(false))
-                {
-                    it.RequiredPrefix = startSlice;
-                    if (it.Seek(it.RequiredPrefix))
-                        throw new InvalidOperationException($"Found attachment stream for attachment key {it.CurrentKey} which should be deleted.");
-                }
+                // There is no way to go from attachment stream to the attachment name.
```

Comment:
We now have debug tag for streams, which might help, but I don't think that this method would do anything for us now that we also have versioning, no?